plugins/pluginmanagers/lazy.nix

```diff
@@ -1,222 +1,166 @@
+
+            keys =
+              with helpers.nixvimTypes;
+              helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
+
+            module = helpers.mkNullOrOption (enum [ false ]) ''
+              Do not automatically load this Lua module when it's required somewhere
+            '';
+
+            priority = helpers.mkNullOrOption number ''
+              Only useful for start plugins (lazy=false) to force loading certain plugins first.
+              Default priority is 50. It's recommended to set this to a high number for colorschemes.
+            '';
+
+            optional = helpers.defaultNullOpts.mkBool false ''
+              When a spec is tagged optional, it will only be included in the final spec,
+              when the same plugin has been specified at least once somewhere else without optional.
+              This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part
+              of the user's plugins
+            '';
+
+            opts =
+              with helpers.nixvimTypes;
+              helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
+                opts should be a table (will be merged with parent specs),
+                return a table (replaces parent specs) or should change a table.
+                The table will be passed to the Plugin.config() function.
+                Setting this value will imply Plugin.config()
+              '';
           };
-      processDependencies =
-        if plugin ? dependencies && plugin.dependencies != null then
-          builtins.concatMap processPlugin plugin.dependencies
-        else
-          [ ];
-    in
-    [ (mkEntryFromDrv plugin) ] ++ processDependencies;
+        });
 
-  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
-  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
-in
-{
-  options = {
-    plugins.lazy = {
-      enable = mkEnableOption "lazy.nvim";
+      listOfPlugins = types.listOf lazyPluginType;
+    in
+    {
+      gitPackage = lib.mkPackageOption pkgs "git" { nullable = true; };
 
-      gitPackage = lib.mkPackageOption pkgs "git" {
-        nullable = true;
+      plugins = mkOption {
+        type = listOfPlugins;
+        default = [ ];
+        description = "List of plugins";
       };
+    };
 
-      plugins =
-        with types;
-        let
-          pluginType = either package (submodule {
-            options = {
-              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
+  extraConfig =
+    cfg:
+    let
+      lazyPlugins = cfg.plugins;
 
-              pkg = mkOption {
-                type = package;
-                description = "Vim plugin to install";
+      specs =
+        let
+          pluginToSpec =
+            plugin:
+            if isDerivation plugin then
+              { dir = "${lib.getName plugin}"; }
```

Comment:
I wonder if we're experiencing something like https://github.com/NixOS/nixpkgs/issues/335016

I need to look into this when I have time to delve into it. Maybe ask around on some matrix rooms or discourse forums.

Reply:
> I guess all that's left to get this PR in, would be to update the nixpkgs in the flake.lock file

Glad you got to the bottom of it, I was really scratching my head!

If so, we do that (fairly) regularly so feel free to do any other cleanup in the meantime. Let me know when you'd like another review.

You're welcome to bump the flake.lock temporarily in this PR (but we won't merge it like that, instead this can be revised once we've done a separate update PR).

Note that bumping nixpkgs currently has other issues elsewhere in nixvim.